packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -137,12 +137,26 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
         decoratorsMeta;
     const proto = Ctor.prototype;
 
+    const publicProperties = keys(apiFields);
+    const protoDescriptors = Object.getOwnPropertyDescriptors(proto);
+    const privateProperties = [];
+    for (const prop in protoDescriptors) {
+        const propDescriptor = protoDescriptors[prop];
+        if ((propDescriptor.get || propDescriptor.set) && publicProperties.indexOf(prop) === -1)
+            privateProperties.push(prop);
+    }
```

Comment:
This logic for creating the `privateProperties` array is technically only needed in dev mode. It would be great to strip it out in production mode.

Instead of having this logic here, maybe we can pass the `protoDescriptors` into `HTMLBridgeElementFactory` and build the `privateProperties` array within an `if (process.env.NODE_ENV !== 'production')` block?